### PR TITLE
[recommendationservice] - base image, licensing and diagram update

### PR DESCRIPTION
**Pull request by julianocosta89**
# Changes

- Updated base image of recommendationservice to use a smaller one.
  - Image size reduced from 991MB to 190MB.
- As we have changed the files enough, I've updated the licensing to OTel authors.
- Updated demo diagram, as the call from recommendationservice to featureflagservice was missing.

## Merge Requirements

For new features contributions please make sure you have completed the following
essential items:

* [x] `CHANGELOG.md` updated to document new feature additions
* [x] Appropriate documentation updates in the [docs](../docs/) folder

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

### CODE REVIEW

**Comment by cartersocha** [comment]
I'm getting a build error when trying to install psutils. 

 Building wheels for collected packages: psutil
#0 32.64   Building wheel for psutil (setup.py): started
#0 33.53   Building wheel for psutil (setup.py): finished with status 'error'
#0 33.54   error: subprocess-exited-with-error
#0 33.54   
#0 33.54   × python setup.py bdist_wheel did not run successfully.
#0 33.54   │ exit code: 1
#0 33.54   ╰─> [43 lines of output]
#0 33.54       running bdist_wheel
#0 33.54       running build
#0 33.54       running build_py
#0 33.54       creating build
#0 33.54       creating build/lib.linux-aarch64-cpython-310
#0 33.54       creating build/lib.linux-aarch64-cpython-310/psutil
#0 33.54       copying psutil/_psbsd.py -> build/lib.linux-aarch64-cpython-310/psutil
#0 33.54       copying psutil/_psosx.py -> build/lib.linux-aarch64-cpython-310/psutil
#0 33.54       copying psutil/_common.py -> build/lib.linux-aarch64-cpython-310/psutil
#0 33.54       copying psutil/_pswindows.py -> build/lib.linux-aarch64-cpython-310/psutil
#0 33.54       copying psutil/_psposix.py -> build/lib.linux-aarch64-cpython-310/psutil
#0 33.54       copying psutil/__init__.py -> build/lib.linux-aarch64-cpython-310/psutil
#0 33.54       copying psutil/_compat.py -> build/lib.linux-aarch64-cpython-310/psutil
#0 33.54       copying psutil/_pssunos.py -> build/lib.linux-aarch64-cpython-310/psutil
#0 33.54       copying psutil/_pslinux.py -> build/lib.linux-aarch64-cpython-310/psutil
#0 33.54       copying psutil/_psaix.py -> build/lib.linux-aarch64-cpython-310/psutil
#0 33.54       creating build/lib.linux-aarch64-cpython-310/psutil/tests
#0 33.54       copying psutil/tests/test_osx.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 33.54       copying psutil/tests/test_unicode.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 33.54       copying psutil/tests/test_connections.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 33.54       copying psutil/tests/test_posix.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 33.54       copying psutil/tests/test_process.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 33.54       copying psutil/tests/test_contracts.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 33.54       copying psutil/tests/test_misc.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 33.54       copying psutil/tests/__init__.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 33.54       copying psutil/tests/test_sunos.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 33.54       copying psutil/tests/test_aix.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 33.54       copying psutil/tests/test_testutils.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 33.54       copying psutil/tests/test_bsd.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 33.54       copying psutil/tests/test_memleaks.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 33.54       copying psutil/tests/__main__.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 33.54       copying psutil/tests/test_linux.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 33.54       copying psutil/tests/test_system.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 33.54       copying psutil/tests/test_windows.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 33.54       copying psutil/tests/runner.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 33.54       running build_ext
#0 33.54       building 'psutil._psutil_linux' extension
#0 33.54       creating build/temp.linux-aarch64-cpython-310
#0 33.54       creating build/temp.linux-aarch64-cpython-310/psutil
#0 33.54       gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DPSUTIL_POSIX=1 -DPSUTIL_SIZEOF_PID_T=4 -DPSUTIL_VERSION=592 -DPSUTIL_LINUX=1 -DPSUTIL_ETHTOOL_MISSING_TYPES=1 -I/usr/local/include/python3.10 -c psutil/_psutil_common.c -o build/temp.linux-aarch64-cpython-310/psutil/_psutil_common.o
#0 33.54       C compiler or Python headers are not installed on this system. Try to run:
#0 33.54       sudo apt-get install gcc python3-dev
#0 33.54       error: command 'gcc' failed: No such file or directory
#0 33.54       [end of output]
#0 33.54   
#0 33.54   note: This error originates from a subprocess, and is likely not a problem with pip.
#0 33.54   ERROR: Failed building wheel for psutil
#0 33.54   Running setup.py clean for psutil
#0 34.09 Failed to build psutil
#0 34.17 Installing collected packages: wrapt, typing-extensions, six, python-json-logger, python-dotenv, psutil, protobuf, opentelemetry-semantic-conventions, backoff, opentelemetry-proto, grpcio, googleapis-common-protos, deprecated, opentelemetry-api, grpcio-health-checking, opentelemetry-sdk, opentelemetry-instrumentation, opentelemetry-exporter-otlp-proto-grpc, opentelemetry-distro
#0 34.50   Running setup.py install for psutil: started
#0 35.64   Running setup.py install for psutil: finished with status 'error'
#0 35.64   error: subprocess-exited-with-error
#0 35.64   
#0 35.64   × Running setup.py install for psutil did not run successfully.
#0 35.64   │ exit code: 1
#0 35.64   ╰─> [45 lines of output]
#0 35.64       running install
#0 35.64       /usr/local/lib/python3.10/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
#0 35.64         warnings.warn(
#0 35.64       running build
#0 35.64       running build_py
#0 35.64       creating build
#0 35.64       creating build/lib.linux-aarch64-cpython-310
#0 35.64       creating build/lib.linux-aarch64-cpython-310/psutil
#0 35.64       copying psutil/_psbsd.py -> build/lib.linux-aarch64-cpython-310/psutil
#0 35.64       copying psutil/_psosx.py -> build/lib.linux-aarch64-cpython-310/psutil
#0 35.64       copying psutil/_common.py -> build/lib.linux-aarch64-cpython-310/psutil
#0 35.64       copying psutil/_pswindows.py -> build/lib.linux-aarch64-cpython-310/psutil
#0 35.64       copying psutil/_psposix.py -> build/lib.linux-aarch64-cpython-310/psutil
#0 35.64       copying psutil/__init__.py -> build/lib.linux-aarch64-cpython-310/psutil
#0 35.64       copying psutil/_compat.py -> build/lib.linux-aarch64-cpython-310/psutil
#0 35.64       copying psutil/_pssunos.py -> build/lib.linux-aarch64-cpython-310/psutil
#0 35.64       copying psutil/_pslinux.py -> build/lib.linux-aarch64-cpython-310/psutil
#0 35.64       copying psutil/_psaix.py -> build/lib.linux-aarch64-cpython-310/psutil
#0 35.64       creating build/lib.linux-aarch64-cpython-310/psutil/tests
#0 35.64       copying psutil/tests/test_osx.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 35.64       copying psutil/tests/test_unicode.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 35.64       copying psutil/tests/test_connections.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 35.64       copying psutil/tests/test_posix.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 35.64       copying psutil/tests/test_process.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 35.64       copying psutil/tests/test_contracts.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 35.64       copying psutil/tests/test_misc.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 35.64       copying psutil/tests/__init__.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 35.64       copying psutil/tests/test_sunos.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 35.64       copying psutil/tests/test_aix.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 35.64       copying psutil/tests/test_testutils.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 35.64       copying psutil/tests/test_bsd.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 35.64       copying psutil/tests/test_memleaks.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 35.64       copying psutil/tests/__main__.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 35.64       copying psutil/tests/test_linux.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 35.64       copying psutil/tests/test_system.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 35.64       copying psutil/tests/test_windows.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 35.64       copying psutil/tests/runner.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
#0 35.64       running build_ext
#0 35.64       building 'psutil._psutil_linux' extension
#0 35.64       creating build/temp.linux-aarch64-cpython-310
#0 35.64       creating build/temp.linux-aarch64-cpython-310/psutil
#0 35.64       gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DPSUTIL_POSIX=1 -DPSUTIL_SIZEOF_PID_T=4 -DPSUTIL_VERSION=592 -DPSUTIL_LINUX=1 -DPSUTIL_ETHTOOL_MISSING_TYPES=1 -I/usr/local/include/python3.10 -c psutil/_psutil_common.c -o build/temp.linux-aarch64-cpython-310/psutil/_psutil_common.o
#0 35.64       C compiler or Python headers are not installed on this system. Try to run:
#0 35.64       sudo apt-get install gcc python3-dev
#0 35.64       error: command 'gcc' failed: No such file or directory
#0 35.64       [end of output]
#0 35.64   
#0 35.64   note: This error originates from a subprocess, and is likely not a problem with pip.
#0 35.64 error: legacy-install-failure
#0 35.64 
#0 35.64 × Encountered error while trying to install package.
#0 35.64 ╰─> psutil
#0 35.64 
#0 35.64 note: This is an issue with the package mentioned above, not pip.
#0 35.64 hint: See above for output from the failure.

**Comment by cartersocha** [comment]
I think this thread answers the root issue: https://github.com/watson-developer-cloud/python-sdk/issues/418#issuecomment-375740919

Question is do we want to continue to use the slim package and update our build docs or find an image with the dependency

**Comment by mviitane** [comment]
I have the same problem. Doesn't build on Mac M1. 

To simplify a bit, the licensing and diagram changes could be split into another PR.

**Review by mviitane — APPROVED** [review]
Builds and runs fine after that latest change.

**Comment by cartersocha** [comment]
Lgtm. Build time was roughly equivalent but image size dropped from 1.52 GB to 284 MB. Nice!

**Review by cartersocha — APPROVED** [review]

File: src/recommendationservice/Dockerfile [hunk 1]
```diff
@@ -12,21 +12,33 @@
...
+RUN pip install --prefix="/reqs" -r requirements.txt
 
 # add files into working directory
 RUN python -m pip install grpcio-tools==1.48.2
 RUN python -m grpc_tools.protoc -I=./proto/ --python_out=./ --grpc_python_out=./ ./proto/demo.proto
 
+FROM base
+
+WORKDIR /usr/src/app/
+COPY --from=builder /usr/src/app/ .
```
Comment (svrnm) [inline src/recommendationservice/Dockerfile:38]: You should be able to squeeze out some build time by doing the installation of the requirements and the grpc tools run in two separate stages that then can be run in parallel, e.g:

```Dockerfile
FROM base as grpc-builder

WORKDIR /usr/src/app/

COPY ./pb/ ./proto/

# add files into working directory
RUN python -m pip install grpcio-tools==1.48.2
RUN python -m grpc_tools.protoc -I=./proto/ --python_out=./ --grpc_python_out=./ ./proto/demo.proto

FROM base

WORKDIR /usr/src/app/
COPY --from=grpc-builder /usr/src/app/ .
COPY --from=builder /reqs /usr/local
```

Gives me ~10 seconds (from ~35 down to 25)

Comment (julianocosta89) [reply src/recommendationservice/Dockerfile]: Awesome!
Thanks for the tip @svrnm! 🤓

**Pull request merged**